runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -324,9 +344,20 @@ private void updateList(QuestContainer questContainer, String filter)
 			Widget quest = questInfo.getQuest();
 			QuestState questState = QuestState.getByColor(quest.getTextColor());
 
-			boolean hidden = !(filter.isEmpty()
-				|| questInfo.getTitle().contains(filter))
-				|| (filter.isEmpty() && hideCompleted && questState == QuestState.COMPLETE);
+			if (questState == null) continue;
+
+			boolean hidden;
+
+			if (currentFilterState != QuestState.ALL)
```

Comment:
I think this disaster reduces to:

```java
			boolean hidden;

			if (!filter.isEmpty())
			{
				// If searching, show result regardless of filtered state
				hidden = !questInfo.getTitle().contains(filter);
			}
			else
			{
				// Otherwise hide if it doesn't match the filter state
				hidden = !questState.getName().equals(currentFilterState.getName());
			}
```

Reply:
Also, questState vs currentFilterState is just an enum reference comparison

Reply:
Thank you for reducing this disaster, please note that it already was unnecessarily verbose in the initial implementation. I just didn't bother to refactor it properly which was my mistake.